### PR TITLE
Unstable release sync with main

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
TODO:
 - [x] swap ember-apply to globby so that dev node can stay on node 14

Inspired by: https://github.com/ember-animation/ember-animated/blob/master/.github/workflows/publish-unstable.yml

CLI Output:
```
❯ node ./test-packages/unstable-release/version.js 
Setting version of packages/addon-dev
Setting version of packages/addon-shim
Setting version of packages/babel-loader-8
Setting version of packages/compat
Setting version of packages/core
Setting version of packages/hbs-loader
Setting version of packages/macros
Setting version of packages/router
Setting version of packages/shared-internals
Setting version of packages/test-setup
Setting version of packages/util
Setting version of packages/webpack
Updating dependencies of packages/addon-dev
Updating dependencies of packages/addon-shim
Updating dependencies of packages/babel-loader-8
Updating dependencies of packages/compat
Updating dependencies of packages/core
Updating dependencies of packages/hbs-loader
Updating dependencies of packages/macros
Updating dependencies of packages/router
Updating dependencies of packages/shared-internals
Updating dependencies of packages/test-setup
Updating dependencies of packages/util
Updating dependencies of packages/webpack
```

Sample Diff:
```
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
• packages/addon-dev/package.json:1: 
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
│  1 │{                                                 │  1 │{
│  2 │  "name": "@embroider/addon-dev",                 │  2 │  "name": "@embroider/addon-dev",
│  3 │  "version": "3.0.0",                             │  3 │  "version": "3.0.0-unstable.331e133e",
│  4 │  "description": "Utilities for addon authors",   │  4 │  "description": "Utilities for addon authors",
│  5 │  "repository": {                                 │  5 │  "repository": {
│  6 │    "type": "git",                                │  6 │    "type": "git",

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
• packages/addon-dev/package.json:26: 
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
│ 26 │    "test": "jest"                                │ 26 │    "test": "jest"
│ 27 │  },                                              │ 27 │  },
│ 28 │  "dependencies": {                               │ 28 │  "dependencies": {
│ 29 │    "@embroider/core": "^2.0.0",                  │ 29 │    "@embroider/core": "2.1.1-unstable.331e133e",
│ 30 │    "@rollup/pluginutils": "^4.1.1",              │ 30 │    "@rollup/pluginutils": "^4.1.1",
│ 31 │    "assert-never": "^1.2.1",                     │ 31 │    "assert-never": "^1.2.1",
│ 32 │    "fs-extra": "^10.0.0",                        │ 32 │    "fs-extra": "^10.0.0",

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
• packages/addon-shim/package.json:1: 
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
│  1 │{                                                 │  1 │{
│  2 │  "name": "@embroider/addon-shim",                │  2 │  "name": "@embroider/addon-shim",
│  3 │  "version": "1.8.4",                             │  3 │  "version": "1.8.4-unstable.331e133e",
│  4 │  "description": "Make v2 addons work in non-Emb↴ │  4 │  "description": "Make v2 addons work in non-Emb↴
│    │                                  …roider apps.", │    │                                  …roider apps.",
│  5 │  "keywords": [],                                 │  5 │  "keywords": [],
│  6 │  "main": "./src/index.js",                       │  6 │  "main": "./src/index.js",

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
• packages/addon-shim/package.json:17: 
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
│ 17 │  },                                              │ 17 │  },
│ 18 │  "scripts": {},                                  │ 18 │  "scripts": {},
│ 19 │  "dependencies": {                               │ 19 │  "dependencies": {
│ 20 │    "@embroider/shared-internals": "^2.0.0",      │ 20 │    "@embroider/shared-internals": "2.0.0-unstab↴
│    │                                                  │    │                                   …le.331e133e",
│ 21 │    "broccoli-funnel": "^3.0.8",                  │ 21 │    "broccoli-funnel": "^3.0.8",
│ 22 │    "semver": "^7.3.8"                            │ 22 │    "semver": "^7.3.8"
│ 23 │  },                                              │ 23 │  },


```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -72,7 +72,7 @@
     "registry": "https://registry.npmjs.org"
   },
   "volta": {
-    "node": "14.19.3",
```
Comment (NullVoxPopuli) [inline package.json:75]: had to do this to tell yarn to install the dependencies of `@embroider/unstable-release`. Not sure if changing the development node is a good idea, or if there is a way to allow yarn to ignore the root node version some how.

File: test-packages/support/package.json [hunk 2]
```diff
@@ -26,7 +26,10 @@
     "qunit": "^2.16.0",
     "webpack": "^5"
   },
+  "engines": {
+    "node": ">= 18.0.0"
+  },
   "volta": {
-    "extends": "../../package.json"
+    "node": ">= 18.0.0"
```
Comment (NullVoxPopuli) [inline test-packages/support/package.json:33]: if we stick with the root package.json dev node on 18, we don't need this

File: test-packages/unstable-release/publish.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    console.info(`Publishing ${workspace}`);
+    await execaCommand('npm publish --tag=unstable --verbose');
+  }
+}
+
+publish();
+
+async function isPrivate() {
+  let pkg = await packageJson.read();
```
Comment (ef4) [inline test-packages/unstable-release/publish.js:16]: Isn't this a runtime error? `packageJson` comes from nowhere.

File: test-packages/unstable-release/publish.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { execaCommand } from 'execa';
+import { project } from 'ember-apply';
```
Comment (NullVoxPopuli) [inline test-packages/unstable-release/publish.js:2]:
```suggestion
import { project, packageJson } from 'ember-apply';
```

File: test-packages/unstable-release/publish.js (hunk 3 above)
Comment (NullVoxPopuli) [reply test-packages/unstable-release/publish.js:16]: yeah, it was -- fixed

File: yarn.lock [hunk 5]
```diff
@@ -8959,6 +8959,21 @@ execa@^5.0.0, execa@^5.1.1:
     signal-exit "^3.0.3"
     strip-final-newline "^2.0.0"
 
+execa@^7.0.0:
```
Comment (NullVoxPopuli) [inline yarn.lock:8962]: I wanted to use new enough execa for execCommand / native ESM.

Node 14.18 is supported: https://github.com/sindresorhus/execa/blob/main/package.json#L16

**Comment by ef4** [comment]
Publishing wasn't really implemented correctly but I fixed it and ran it locally. See the unstable.73213f2a packages on NPM.

**Pull request merged**